specs/integration/events/conversation_created.spec.ts

```diff
@@ -24,13 +35,50 @@ describe('Event conversation:create', () => {
   beforeEach(async () => {
     workerSpy = sinon.spy(webhookJob.worker, 'processJob')
     queueSpy = sinon.spy(webhookJob.queue, 'add')
+
+    // set up webhooks
+    const listIntegrations: SinonStub                  = sinon.stub(IntegrationsApi.prototype, 'listIntegrations')
+    const createIntegrationWithHttpInfo: SinonStub     = sinon.stub(IntegrationsApi.prototype, 'createIntegrationWithHttpInfo')
+    listIntegrations.returns({ integrations: MOCK_INTEGRATIONS })
+    createIntegrationWithHttpInfo.returns({ 
+      response: { 
+        body: { 
+          integration: { 
+            webhooks: [{ secret: webhookIntegrationSecret}]
+          }
+        }
+      }
+    })
+
+    nock(FAKE_AUTH_HOST)
+      .post(FAKE_AUTH_ENDPOINT)
+      .reply(200, {
+        userId: '123',
+        permissions: [GoodChatPermissions.ADMIN],
+        displayName: 'Jane Doe'
+      })
+    
+    await agent.post('/webhooks/connect')
+      .set('Authorization', 'Bearer xyz')
+      .expect(200);
```

Comment:
```suggestion
    await storeWebhookIntegrationSecret(webhookIntegrationSecret)
```

(for other tests as well)
the `beforeEach` is about setting up the correct state for tests, this is a much simpler setup

Reply:
Something I don't quite like about doing `storeWebhookIntegrationSecret` is that it knows too much about how `/connect` works.

We should be able to change the implementation details of how connect works freely whilst keeping the tests passing.

So for that reason I preferred to exercise the public API instead of the internal implementation details.

Reply:
And this knows even more:
- the api calls made
- the sdk functions
- the auth flow
- the api called
- the response format of both api calls

... all things that could change a lot more easily than a simple function.

these files test that `/trigger` triggers the correct event handler, that's what we care about, and this works on its own.

not to mention it's massive piece of code for very little value.